build-script.py

```diff
@@ -636,6 +637,114 @@ def run_xctests(toolchain, build_dir, multiroot_data_file, release, verbose):
+# Arugment Parsing functions
+
+
+def generate_xcodeproj(args):
+    xcode_gen(config=args.xcconfig_path)
+
+
+def generate_source_code_command(args):
+    try:
+        generate_gyb_files(
+            args.gyb_exec,
+            verbose=args.verbose,
+            add_source_locations=args.add_source_locations,
+        )
+    except subprocess.CalledProcessError as e:
+        printerr("FAIL: Generating .gyb files failed")
+        printerr("Executing: %s" % " ".join(e.cmd))
+        printerr(e.output)
+
+    run_code_generation(
+        toolchain=args.toolchain,
+        build_dir=realpath(args.build_dir),
+        multiroot_data_file=args.multiroot_data_file,
+        release=args.release,
+        verbose=args.verbose,
+        swiftsyntaxbuilder_destination=os.path.join(SWIFTSYNTAXBUILDER_DIR, "generated")
+    )
+
+
+def build_command(args):
+    try:
+        builder = Builder(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            verbose=args.verbose,
+            disable_sandbox=args.disable_sandbox,
+        )
+        # Until rdar://53881101 is implemented, we cannot request a build of multiple
+        # targets simultaneously. For now, just build one product after the other.
+        builder.build("SwiftSyntax")
+        builder.build("SwiftSyntaxParser")
+        builder.build("SwiftSyntaxBuilder")
+        builder.build("SwiftSyntaxBuilderGeneration")
+    except subprocess.CalledProcessError as e:
+        printerr("FAIL: Building SwiftSyntax failed")
+        printerr("Executing: %s" % " ".join(e.cmd))
+        printerr(e.output)
+
+
+def test_command(args):
+    try:
+        builder = Builder(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            verbose=args.verbose,
+            disable_sandbox=args.disable_sandbox,
+        )
+        
+        builder.build("lit-test-helper")
+        
+        success = run_tests(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            filecheck_exec=realpath(args.filecheck_exec),
+            skip_lit_tests=args.skip_lit_tests,
+            verbose=args.verbose,
+        )
+        if not success:
+            # An error message has already been printed by the failing test
+            # suite
+            sys.exit(1)
+        else:
+            print("** All tests passed **")
+    except subprocess.CalledProcessError as e:
+        printerr("FAIL: Running tests failed")
+        printerr("Executing: %s" % " ".join(e.cmd))
+        printerr(e.output)
+        sys.exit(1)
+
+def verify_source_code_command(args):
```

Comment:
Nitpick: I think it would be nice if we defined `verify_source_code_command` right below `generate_source_code_command` because they perform very similar tasks. Similarly, I think defining the `verify-source-code` subcommand right after `generate-source-code` would make sense to me.